src/react/components.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import { Realm } from "../realm.js";
+import { ECMAScriptSourceFunctionValue, AbstractValue, ObjectValue, AbstractObjectValue } from "../values/index.js";
+import { flowAnnotationToObjectTypeTemplate } from "../flow/utils.js";
+import * as t from "babel-types";
+import type { BabelNodeIdentifier } from "babel-types";
+import { createAbstractObject } from "../flow/abstractObjectFactories.js";
+import { valueIsClassComponent } from "./utils";
+import { ExpectedBailOut } from "./reconcilation.js";
+import { Get } from "../methods/index.js";
+import { Properties } from "../singletons.js";
+import invariant from "../invariant.js";
+
+export function getInitialProps(
+  realm: Realm,
+  componentType: ECMAScriptSourceFunctionValue
+): ObjectValue | AbstractObjectValue {
+  let propsName = null;
+  let propTypes = null;
+  if (valueIsClassComponent(realm, componentType)) {
+    propsName = "this.props";
```

Comment:
I wonder if `this...` is going to be always bound to the right thing in the code generation.

Reply:
It seems to work in all the tests.